### PR TITLE
Added additional tables and views to the report database

### PR DESCRIPTION
**Pull request by mackdk**
## What does this PR change?

This PR improves the reporting database schema by adding additional tables and views

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- No tests: no code changes

- [X] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/16564

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/common/db/datasource/xml/ChannelReport_queries.xml [hunk 1]
```diff
@@ -20,23 +20,76 @@
...
+                    , rhnerrata.id AS errata_id
+                    , rhnerrata.advisory_name
+            FROM rhnchannelerrata
+                    INNER JOIN rhnchannel ON rhnchannelerrata.channel_id = rhnchannel.id
+                    INNER JOIN rhnerrata ON rhnchannelerrata.errata_id = rhnerrata.id
+        ORDER BY channel_id, errata_id OFFSET :offset LIMIT :limit
+    </query>
+</mode>
+
+<mode name="ChannelContentSource" class="">
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/common/db/datasource/xml/ChannelReport_queries.xml:78]: I would like to not name repositories "ContentSource" in the reportdb.

Can we rename it to "Repository"? It is a more common name so everybody knows what he can expect.
Of cause we cannot rename this in the main DB, but when we introduce this in the reportdb we should use better names if we do not like the original once.

File: java/code/src/com/redhat/rhn/common/db/datasource/xml/GeneralReport_queries.xml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          SELECT rhnUserServerGroupPerms.server_group_id AS system_group_id
+                    , rhnUserServerGroupPerms.user_id
+                    , rhnServerGroup.name AS group_name
+            FROM rhnUserServerGroupPerms
+                    INNER JOIN rhnServerGroup ON rhnUserServerGroupPerms.server_group_id = rhnServerGroup.id
+        ORDER BY system_group_id, user_id OFFSET :offset LIMIT :limit
+    </query>
+</mode>
+
+<mode name="WebUser" class="">
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/common/db/datasource/xml/GeneralReport_queries.xml:42]: What about naming this just "User" ?

File: java/code/src/com/redhat/rhn/common/db/datasource/xml/GeneralReport_queries.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            FROM web_contact
+                    INNER JOIN web_customer ON web_contact.org_id = web_customer.id
+                    INNER JOIN web_user_personal_info ON web_contact.id = web_user_personal_info.web_user_id
+                    INNER JOIN rhnUserInfo ON web_contact.id = rhnUserInfo.user_id
+                    LEFT JOIN latest_state_change ON web_contact.id = latest_state_change.web_contact_id
+        ORDER BY user_id OFFSET :offset LIMIT :limit
+    </query>
+</mode>
+
+<mode name="WebUserGroup" class="">
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/common/db/datasource/xml/GeneralReport_queries.xml:72]: Just "UserGroup"

File: schema/reportdb/common/tables/ChannelContentSource.sql [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+(
+    mgm_id                    NUMERIC NOT NULL,
+    channel_id                NUMERIC NOT NULL,
+    content_source_id         NUMERIC NOT NULL,
+    label                     VARCHAR(128),
+    url                       VARCHAR(100),
+    type                      VARCHAR(64),
+    metadata_signed           BOOLEAN,
+    synced_date               TIMESTAMPTZ DEFAULT (current_timestamp)
+);
```
Comment (mcalmer) [inline schema/reportdb/common/tables/ChannelContentSource.sql:22]: This looks like a join table, but it is not. I think content_source_id exists only to make the primary key unique.

I would suggest to rename this table to "Repository" and maybe the column content_source_id to repository_id.

File: schema/reportdb/common/tables/System.sql [hunk 5]
```diff
@@ -24,6 +24,8 @@ CREATE TABLE System
     last_checkin_time             TIMESTAMPTZ,
     kernel_version                VARCHAR(64),
     architecture                  VARCHAR(64),
+    is_proxy                      BOOLEAN NOT NULL DEFAULT FALSE,
+    proxy_system_id               NUMERIC,
```
Comment (mcalmer) [inline schema/reportdb/common/tables/System.sql:28]: we should also add "is_mgr_server".

File: schema/reportdb/common/tables/WebUser.sql [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+
+CREATE TABLE WebUser (
```
Comment (mcalmer) [inline schema/reportdb/common/tables/WebUser.sql:12]: Suggestion: rename to just "User"

File: schema/reportdb/common/tables/WebUserGroup.sql [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+
+CREATE TABLE WebUserGroup (
```
Comment (mcalmer) [inline schema/reportdb/common/tables/WebUserGroup.sql:12]: Suggestion: rename to "UserGroup"

File: schema/reportdb/common/views/CustomChannelsReport.sql [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+             , Channel.organization
+             , Channel.channel_id
+             , Channel.label
+             , Channel.name
+             , Channel.summary
+             , Channel.description
+             , Channel.parent_channel_label
+             , Channel.arch
+             , Channel.checksum_type
+             , sources.channel_sources
```
Comment (mcalmer) [inline schema/reportdb/common/views/CustomChannelsReport.sql:28]: Suggestion: rename it so "channel_repositories"

File: schema/reportdb/common/views/SystemHistoryKickstartReport.sql [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+--
+-- This software is licensed to you under the GNU General Public License,
+-- version 2 (GPLv2). There is NO WARRANTY for this software, express or
+-- implied, including the implied warranties of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+-- along with this software; if not, see
+-- http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+--
+
+CREATE OR REPLACE VIEW SystemHistoryKickstartReport AS
```
Comment (mcalmer) [inline schema/reportdb/common/views/SystemHistoryKickstartReport.sql:12]: Is this about Kickstart only, or does it cover also autoyast installations? If the latter, then we should rename it to "SystemHistoryAutoinstallationReport"

Comment (mcalmer) [reply schema/reportdb/common/views/SystemHistoryKickstartReport.sql:12]: But didn't we say, that we may not want this ? I do not remember anymore the list.

Comment (mackdk) [reply schema/reportdb/common/views/SystemHistoryKickstartReport.sql]: Renamed the view to SystemHistoryAutoinstallationReport. Other kickstart related reports are indeed low priority and not included.

File: schema/reportdb/common/tables/WebUser.sql (hunk 6 above)
Comment (mackdk) [reply schema/reportdb/common/tables/WebUser.sql]: Renamed to Account as "user" is used by postgres as well.

File: schema/reportdb/common/tables/WebUserGroup.sql (hunk 7 above)
Comment (mackdk) [reply schema/reportdb/common/tables/WebUserGroup.sql]: Renamed to AccountGroup

File: schema/reportdb/common/tables/ChannelContentSource.sql (hunk 4 above)
Comment (mackdk) [reply schema/reportdb/common/tables/ChannelContentSource.sql]: It was meant to be a join table actually but it was duplicating the repository data. So I introduced a new Repository table and renamed this one to ChannelRepository

File: schema/reportdb/common/views/CustomChannelsReport.sql (hunk 8 above)
Comment (mackdk) [reply schema/reportdb/common/views/CustomChannelsReport.sql]: Done

File: java/code/src/com/redhat/rhn/common/db/datasource/xml/ChannelReport_queries.xml (hunk 1 above)
Comment (mackdk) [reply java/code/src/com/redhat/rhn/common/db/datasource/xml/ChannelReport_queries.xml]: Every reference to ContentSource has been renamed to Repository.

**Review by mcalmer — APPROVED** [review]

**Pull request merged**